src/openfiles.ts

```diff
@@ -201,13 +202,23 @@ export class OpenFiles implements vscode.TreeDataProvider<TreeItemFile|TreeItemG
 
 	async getChildrenFromGroup(group: TreeItemGroup): Promise<(TreeItemFile)[]> {
 		let result: TreeItemFile[] = [];
-		for (let document of this.textDocuments) {
-			if (document.languageId !== group.label || document.isClosed) {
-				continue; 
+
+		const documentsByName = {};
+		for (const document of this.textDocuments) {
```

Comment:
The group conditional (that's now on line 215) should live here. Otherwise you're looping to a number of documents that could've been filtered out already, for no real reason.